typed-racket-lib/typed-racket/utils/mutated-vars.rkt

```diff
@@ -17,8 +19,18 @@
         (loop stx tbl)))
     (syntax-parse stx
       #:literal-sets (kernel-literals)
+      ;; let us care about custom syntax classes
+      [form
```

Comment:
Yes, this is used to add the name of the accessor to the mutated variable environment. Since the case the extra `FieldPE:` case above looks at uses of the accessor name, that prevents refinement of the accessor result.